src/Codeception/Util/Template.php

```diff
@@ -63,10 +63,16 @@ public function produce()
 
         foreach ($matches as $match) { // fill in placeholders
             $placeholder = $match[1];
-            if (!isset($this->vars[$placeholder])) {
-                continue;
+            $value = $this->vars;
+            foreach (explode('.', $placeholder) as $segment) {
+                if (is_array($value) && array_key_exists($segment, $value)) {
```

Comment:
Since it's used a lot it could be adjusted as `is_array($value) && isset($value[$segment]) && array_key_exists($segment, $value)` to get extra performance. Needs to be measured though...